statsmodels/tsa/innovations/_arma_innovations.pyx.in

```diff
@@ -0,0 +1,734 @@
+#cython: boundscheck=False
+#cython: wraparound=False
+#cython: cdivision=False
+"""
+Innovations algorithm
+
+Author: Chad Fulton
+License: Simplified-BSD
+"""
+
+{{py:
+
+TYPES = {
+    "s": ("np.float32_t", "np.float32", "np.NPY_FLOAT32"),
+    "d": ("np.float64_t", "float", "np.NPY_FLOAT64"),
+    "c": ("np.complex64_t", "np.complex64", "np.NPY_COMPLEX64"),
+    "z": ("np.complex128_t", "complex", "np.NPY_COMPLEX128"),
+}
+
+}}
+
+# Typical imports
+import numpy as np
+import warnings
+from statsmodels.tsa import arima_process
+from statsmodels.src.math cimport *
+cimport numpy as np
+cimport cython
+
+np.import_array()
+
+cdef int C = 0
+
+
+{{for prefix, types in TYPES.items()}}
+{{py:cython_type, dtype, typenum = types}}
+{{py:
+combined_prefix = prefix
+combined_cython_type = cython_type
+if prefix == 'c':
+    combined_prefix = 'z'
+    combined_cython_type = 'np.complex128_t'
+if prefix == 's':
+    combined_prefix = 'd'
+    combined_cython_type = 'np.float64_t'
+}}
```

Comment:
looks like a good use case for fused types!

Reply:
Agree.  Fused types + pyx allows for coverage checks.

Reply:
Good idea - not for initial commit, but I do like the idea of moving to fused types in the future.